plugins/single-instance/src/platform_impl/macos.rs

```diff
@@ -4,13 +4,164 @@
+                        // -> Notify singleton of launch and exit.
+                        match notify_singleton(socket) {
+                            Ok(_) => {
+                                std::process::exit(0);
+                            }
+                            Err(ErrorKind::Retry) => {
+                                start(app, cb, true);
+                            }
+                            Err(ErrorKind::Abort) => {
+                                eprintln!("single_instance failed - launching normally");
+                            }
+                        }
+                    } else {
+                        eprintln!(
+                            "single_instance failed - launching normally: address already in use"
+                        );
+                    }
+                }
+                Some(e) => {
+                    // Unhandled OS error -> Try to launch instead.
+                    eprintln!("single_instance failed - launching normally: {}", e);
+                }
+                None => {
+                    // Not an OS error -> Try to launch instead
+                    eprintln!("single_instance failed - launching normally: {}", err);
+                }
+            }
+        }
+    };
 }
 
-pub fn destroy<R: Runtime, M: Manager<R>>(_manager: &M) {}
+fn listen_for_other_instances<A: Runtime>(
+    listener: UnixListener,
+    app: AppHandle<A>,
+    mut cb: Box<SingleInstanceCallback<A>>,
+) {
+    let cwd = std::env::current_dir()
+        .unwrap_or_default()
+        .to_str()
+        .unwrap_or_default()
+        .to_string();
+
+    tauri::async_runtime::spawn(async move {
+        for stream in listener.incoming() {
+            match stream {
+                Ok(mut stream) => {
+                    let mut s = String::new();
+                    match stream.read_to_string(&mut s) {
+                        Ok(_) => {
+                            let args: Vec<String> = s.split('\0').map(String::from).collect();
+                            cb(&app.clone().app_handle(), args, cwd.clone());
+                        }
+                        Err(e) => println!("Error reading stream: {e}"),
+                    }
+                }
+                Err(err) => {
+                    eprintln!("single_instance notify failed: {}", err);
+                    continue;
+                }
+            }
+        }
+    });
+}
+
+fn notify_singleton(socket: PathBuf) -> Result<(), ErrorKind> {
+    match UnixStream::connect(&socket) {
+        Ok(stream) => {
+            let mut bf = BufWriter::new(&stream);
+            let args_joined = std::env::args().collect::<Vec<String>>().join("\0");
+
+            if let Err(_e) = bf.write_all(args_joined.as_bytes()) {
+                return Err(ErrorKind::Abort);
+            }
+            if let Err(_e) = bf.flush() {
+                return Err(ErrorKind::Abort);
+            }
```

Comment:
hmm, i feel like `.map_err(|_| ErrorKind::Abort)?` may look a bit cleaner. What do you think?